rotkehlchen/tests/api/test_balances.py

```diff
@@ -973,3 +983,108 @@ def query_blockchain_balance(num: int) -> Any:
+def test_query_balances_with_threshold(
+        rotkehlchen_api_server_with_exchanges: 'APIServer',
+        ethereum_accounts: list['ChecksumEvmAddress'],
+        btc_accounts: list['BTCAddress'],
+) -> None:
+    """Test that balance filtering by USD value threshold works for all balance types"""
+    rotki = rotkehlchen_api_server_with_exchanges.rest_api.rotkehlchen
+
+    manually_tracked_balances = [
+        ManuallyTrackedBalance(
+            identifier=-1,
+            asset=A_EUR,
+            label='Small EUR',
+            amount=ONE,
+            location=Location.BANKS,
+            tags=None,
+            balance_type=BalanceType.ASSET,
+        ),
+        ManuallyTrackedBalance(
+            identifier=-2,
+            asset=A_USD,
+            label='USD',
+            amount=FVal('10'),
+            location=Location.BANKS,
+            tags=None,
+            balance_type=BalanceType.ASSET,
+        ),
+        ManuallyTrackedBalance(
+            identifier=-3,
+            asset=A_BTC,
+            label='Large BTC',
+            amount=ONE,
+            location=Location.EXTERNAL,
+            tags=None,
+            balance_type=BalanceType.ASSET,
+        ),
+    ]
+
+    setup = setup_balances(
+        rotki=rotki,
+        ethereum_accounts=ethereum_accounts,
+        btc_accounts=btc_accounts,
+        manually_tracked_balances=manually_tracked_balances,
+        eth_balances=['100000000000000000', '200000000000000000'],  # 0.1 and 0.2 ETH
+        token_balances={
+            A_DAI.resolve_to_evm_token(): ['5000000000000000000', '0'],  # 5 DAI and 0 DAI
+            A_USDC.resolve_to_evm_token(): ['1000000', '2000000'],  # 1 USDC and 2 USDC
+        },
+    )
+
+    threshold = 10  # Set threshold to 10 USD
+
+    with ExitStack() as stack:
+        setup.enter_all_patches(stack)
+
+        # Test blockchain balances
+        response = requests.get(
+            api_url_for(
+                rotkehlchen_api_server_with_exchanges,
+                'blockchainbalancesresource',
+            ),
+            params={'usd_value_threshold': threshold},
+        )
+        blockchain_result = assert_proper_sync_response_with_result(response)
+
+        # Test exchange balances
+        response = requests.get(
+            api_url_for(
+                rotkehlchen_api_server_with_exchanges,
+                'exchangebalancesresource',
+            ),
+            params={'usd_value_threshold': threshold},
+        )
+        exchange_result = assert_proper_sync_response_with_result(response)
+
+        # Test manual balances
+        response = requests.get(
+            api_url_for(
+                rotkehlchen_api_server_with_exchanges,
+                'manuallytrackedbalancesresource',
+            ),
+            params={'usd_value_threshold': threshold},
+        )
+        manual_result = assert_proper_sync_response_with_result(response)
```

Comment:
Maybe you can do this cleaner with a for loop?

```suggestion
        results = []
        for endpoint in (
            'blockchainbalancesresource',
            'exchangebalancesresource',
            'manuallytrackedbalancesresource'
        ):
            response = requests.get(
                api_url_for(
                     rotkehlchen_api_server_with_exchanges,
                    endpoint,
               ),
               params={'usd_value_threshold': threshold},
           )
           results.append(assert_proper_sync_response_with_result(response))
       
       blockchain_result, exchange_result, manual_result = results
```